### PR TITLE
Pivotal hotfix creation from the cli

### PR DESCRIPTION
**Pull request by jhirbour**
originally from pull/106#issuecomment-102415174
# What it does

We created a script that will create a bug pivotal story and start a hotfix branch for you. All you need to do is pass a story name.
# Setting it up

If you want to use it, you have to add your Pivotal API Key as an environment variable. On my account, I put it in ~/.zsh/control-rails.zsh. (but all files in ~/.zsh our loaded so put it any file in that directory)

i.e. 

```
export PIVOTAL_TOKEN="TOKEN HERE"
export PIVOTAL_PROJECT_ID="some_number_here"
```

You can find your token at the bottom of the User Profile page on Pivotal, which you can get to by clicking your name in the top right corner.

Before using, make sure to source the control-rails file so the variable loads in, or start a new session.
# How you'd use it

i.e. script/create-hotfix 'Do The Needful'
# What it does

It will detect who you are hitched with and add both as story owners. It won't run if you have unstaged git changes. It will then create the branch name using the slugify pivotal card bookmarklet logic (thanks Casey!).
# Misc
- This also adds the .bin directory in this repo to your path.
- This also moves git-flow-rebase from the control repo into this repo (as it's not rails related)

### CODE REVIEW

File: home/.bin/create-hotfix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+hitch_config = YAML.load_file("#{ENV['HOME']}/.hitchrc")
+story_owners = if hitch_config[:current_pair].empty?
+                 Array(ENV['USER'])
+               else
+                 hitch_config[:current_pair]
+               end
+
+client = TrackerApi::Client.new(token: pivotal_token)
+# we only have one project so this works well
+control = client.project(project_id)
```
Comment (benjaminoakes) [inline home/.bin/create-hotfix]: Should probably rename `control` to `project` or something similar.

File: home/.zshrc [hunk 2]
```diff
@@ -90,6 +90,7 @@ echo "-----"
 
 PATH=$PATH:$HOME/.rvm/bin # Add RVM to PATH for scripting
 PATH=$PATH:$HOME/bin # Make personal scripts available
+PATH=$PATH:$HOME/.homesick/repos/dotfiles/home/.bin # Make personal scripts available
```
Comment (benjaminoakes) [inline home/.zshrc:93]: Isn't this symlinked?

File: home/.bin/create-hotfix [hunk 3]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env ruby
```
Comment (benjaminoakes) [inline home/.bin/create-hotfix:1]: We should add your documentation from the pull request as comments here.  With any luck, others will be able to use this script and (hopefully) contribute changes  back to us.

File: home/.bin/create-hotfix [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+# i.e.
+#    export PIVOTAL_TOKEN="TOKEN HERE"
+#    export PIVOTAL_PROJECT_ID="some_number_here"
+#
+#    You can find your token at the bottom of the User Profile page on Pivotal,
+#    which you can get to by clicking your name in the top right corner.
+#
+# How you'd use it
+#     i.e. script/create-hotfix 'Do The Needful'
```
Comment (treznick) [inline home/.bin/create-hotfix:28]: is this still `script/create-hotfix` or being in `.bin` is it just `create-hotfix`?

File: home/.bin/create-hotfix [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+# we only have one project so this works well
+project = client.project(project_id)
+
+pivotal_users = project.memberships.map(&:person).select do |person|
+  story_owners.include? person.email.split("@").first
+end
+
+pivotal_ids = pivotal_users.map(&:id)
+
+story = project.create_story(name: story_name, story_type: 'bug', current_state: 'started', owner_ids: pivotal_ids)
```
Comment (treznick) [inline home/.bin/create-hotfix:81]: Hrm, we can create other story types...? Hrm.... methinks there may be a PR for some REPL like stuff with start story...

File: home/.bin/create-hotfix (hunk 4 above)
Comment (jhirbour) [inline home/.bin/create-hotfix:28]: good catch.

File: home/.zshrc (hunk 2 above)
Comment (jnimety) [inline home/.zshrc:93]: this should be `$HOME/.bin`, the repo name is not guaranteed to be `dotfiles`

Comment (jnimety) [inline home/.zshrc:93]: done

**Pull request merged**